docs/proposals/karmada-operator/api-server-sidecard-containers/README.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Support for API Server Sidecar Containers in Karmada Operator
+authors:
+- "@jabellard"
+reviewers:
+- "@RainbowMango"
+approvers:
+- "@RainbowMango"
+
+creation-date: 2025-02-07
+
+---
+
+# Support for API Server Sidecar Containers in Karmada Operator
+
+## Summary
+
+This proposal aims to enhance the Karmada operator by introducing support for configuring additional containers  for the Karmada API server component.
+By allowing users to define additional containers, this feature unlocks new use cases such as integration with Key Management Service (KMS) plugins for configuring encryption at rest. 
+This enhancement provides greater flexibility while maintaining the declarative nature of the Karmada operator’s deployment model.
+
+## Motivation
+
+Currently, the Karmada operator provisions the API server as a standalone container within a pod. While the API server supports configurability via extra arguments, additional volumes, and volume mounts, it lacks built-in support for defining additional containers.
+By introducing explicit support for additional containers, this feature enables:
+
+- Seamless integration of auxiliary services such as KMS-based encryption providers.
+- Greater deployment flexibility by enabling users to extend the API server’s functionality without modifying the core operator logic.
+
+### Goals
+- Introduce support for configuring sidecar containers for the Karmada API server.
+
+### Non-Goals
+
+- Introduce changes beyond the scope of what is needed to support additional containers for the Karmada API server component.
+
+### API Changes
+
+```go
+// KarmadaAPIServer holds settings for the Karmada API server.
+type KarmadaAPIServer struct {
+    // Other, existing fields omitted for brevity
```

Comment:
I need to check if we need to add further comments to the existing fields, to avoid potential ambiguity.
Maybe after adding an example(that I asked below :) )